core/textinput/src/textinput/Editor.cpp

```diff
@@ -167,9 +186,10 @@ namespace textinput {
   Editor::ProcessChar(char C, EditorRange& R) {
     if (C < 32) return kPRError;
 
-    if (fMode == kHistSearchMode) {
+    if (fMode == kHistRevSearchMode || fMode == kHistFwdSearchMode) {
       fSearch += C;
-      SetReverseHistSearchPrompt(R.fDisplay);
+      if (fMode == kHistFwdSearchMode) SetForwardHistSearchPrompt(R.fDisplay);
+      else SetReverseHistSearchPrompt(R.fDisplay);
```

Comment:
With the above change this becomes
```suggestion
      SetHistSearchModePrompt(R.fDisplay);
```